### ISSUE TITLE
Problem while profiling the code

### ISSUE THREAD

**Issue by CarstenF** [safarijv/yet-another-django-profiler#30]
I serve my Django project with Apache via mod_wsgi and attempt to use YADP for profiling a specific view.

The view is known to work flawlessly without profiling, i.e. without the `profile` parameter added to the HTTP GET request. It also works flawlessly if the `profile` parameter is given but YADP is omitted from the `MIDDLEWARE_CLASSES`. Also, I employ another, independent profiling strategy (only alternatingly with YADP, of course!), namely https://pypi.python.org/pypi/repoze.profile along with kcachegrind, for verification and comparison.

The problem is that my view's main loop should iterate 20 times, but I can see from the profile results that, when profiled with YADP, something happens during the first iteration of the loop that probably throws an exception and thus stops the profiling.

The problem is that (besides subtly unexpected profile results) there is no indication that an exception has occurred. It seems that learning what the exception was and thus what the cause the problem was is even more difficult -- I didn't manage it. (Although the issue itself can easily be reproduced: Just insert an `assert False` into the view's code.)

In short:
I experience exceptions of unknown type and cause when views are profiled with YADP. What can I do?

**Comment by CarstenF** [comment]
Well, the Apache error log states the details about the exception:

`TransactionManagementError: select_for_update cannot be used outside of a transaction.`

I have however no idea at all how or why YADP interferes with transaction handling and how it can cause my setting

`DATABASES["default"]["ATOMIC_REQUESTS"] = True`

to be disregarded…
